code/modules/reagents/chemistry/reagents/misc.dm

```diff
@@ -421,6 +421,32 @@
 /datum/reagent/love/reaction_mob(mob/living/M, method=TOUCH, volume)
 	to_chat(M, "<span class='notice'>You feel loved!</span>")
 
+/datum/reagent/jestosterone //Formerly known as Nitrogen tungstide hypochlorite before NT fired the chemists for trying to be funny
+	name = "Jestosterone"
+	id = "jestosterone"
+	description = "Jestosterone is an odd chemical compound that induces a variety of annoying side-effects in the average person. It also causes mild intoxication."
+	color = "#ff00ff" //Fuchsia, pity we can't do rainbow here
+	taste_message = "a funny flavour"
+
+/datum/reagent/jestosterone/reaction_mob(mob/living/M, method, volume)
+	..()
+	if(istype(M, /mob/living/carbon))
+		return
+	var/is_a_clown = FALSE
+	if(M.mind && (M.mind.assigned_role == "Clown" || M.mind.assigned_role == SPECIAL_ROLE_HONKSQUAD))
+		is_a_clown = TRUE
+	if(!is_a_clown)
+		M.AdjustDizzy(volume)
+	M.AddComponent(/datum/component/jestosterone, M, is_a_clown)
+
+/datum/reagent/jestosterone/on_mob_life(mob/living/M)
+	M.SendSignal(COMSIG_ON_MOB_LIFE, M)
```

Comment:
:eyes:

Reply:
(if you *were* to implement this, this should be a general component, not individually fired on a single reagent. but reagents already have a method of doing stuff on mob life, so use that instead)